### PR TITLE
Stack: Moving AzureRm.storage module to consume api-version 2016-01-01

### PR DESCRIPTION
**Pull request by bganapa**
## Description
 AzureRM.Storage module version update to 2016-01-01                                                              
    Copying Storage SDK version 5.0.0-preview                                                                               
    Corresponding cmdlet changes, redoing IStorageContext move and Common runtime move, Redoing HashTable[] to HashTable    

    Compute changes for consuming the storage sdk api-version 2016-01-01                                                    

   Fixing Azure.Storage module for IStorageContext changes, comparing to the latest azure version 

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Compute/Stack/Commands.Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs [hunk 1]
```diff
@@ -209,16 +210,17 @@ protected string GetStorageKey(string storageName)
 
             if (!string.IsNullOrEmpty(storageName))
             {
-                var storageClient = new StorageManagementClient();
+                var storageClient = AzureSession.Instance.ClientFactory.CreateArmClient<StorageManagementClient>(DefaultProfile.DefaultContext,
```
Comment (bganapa) [inline src/ResourceManager/Compute/Stack/Commands.Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs]: remove extra whitespaces

File: src/ResourceManager/Compute/Stack/Compute.sln [hunk 2]
```diff
@@ -104,6 +104,10 @@ Global
 		{1338F7AE-7111-4ED3-8916-2D0FECC876F4}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{1338F7AE-7111-4ED3-8916-2D0FECC876F4}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{1338F7AE-7111-4ED3-8916-2D0FECC876F4}.Release|Any CPU.Build.0 = Release|Any CPU
+		{37C44181-3F1B-4ABD-8089-26DFAB4B6BA8}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (bganapa) [inline src/ResourceManager/Compute/Stack/Compute.sln]: This change should not be needed

File: src/ResourceManager/Storage/Stack/Commands.Management.Storage.Test/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -47,5 +47,6 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 
-[assembly: AssemblyVersion("1.0.4")]
-[assembly: AssemblyFileVersion("1.0.4")]
+[assembly: AssemblyVersion("1.0.6")]
```
Comment (bganapa) [inline src/ResourceManager/Storage/Stack/Commands.Management.Storage.Test/Properties/AssemblyInfo.cs]: change this to 1.1.0

File: src/ResourceManager/Storage/Stack/Commands.Management.Storage/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -46,5 +46,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 
-[assembly: AssemblyVersion("3.3.2")]
-[assembly: AssemblyFileVersion("3.3.2")]
+[assembly: AssemblyVersion("1.1.0")]
```
Comment (bganapa) [inline src/ResourceManager/Storage/Stack/Commands.Management.Storage/Properties/AssemblyInfo.cs]: Recheck here , looks like the version is being downgraded?

File: src/ResourceManager/Storage/Stack/Commands.Management.Storage/help/AzureRM.Storage.md [hunk 5]
```diff
@@ -1,40 +0,0 @@
----
```
Comment (bganapa) [inline src/ResourceManager/Storage/Stack/Commands.Management.Storage/help/AzureRM.Storage.md]: Need to add Help MD files

**Comment by vladimir-shcherbakov** [comment]
@bganapa, One of the CI-build failed, Could you take a look?

**Comment by deathly809** [comment]
@vladimir-shcherbakov I think it might be a CI issue, 

[31;1mThe command "sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys BE1229CF" failed and exited with 2 during .[0m

File: src/ResourceManager/Compute/Stack/Commands.Compute.Test/Commands.Compute.Test.csproj [hunk 6]
```diff
@@ -188,7 +188,7 @@
       <Project>{3436a126-edc9-4060-8952-9a1be34cdd95}</Project>
       <Name>Commands.ScenarioTests.ResourceManager.Common</Name>
     </ProjectReference>
-    <ProjectReference Include="..\..\..\..\Common\Commands.Common.Storage\Stack\Commands.Common.Storage.csproj">
+    <ProjectReference Include="..\..\..\..\Common\Commands.Common.Storage\Commands.Common.Storage.csproj">
```
[the remark this answers is not in the document]
Comment (bganapa) [reply src/ResourceManager/Compute/Stack/Commands.Compute.Test/Commands.Compute.Test.csproj]: reverted

File: src/ResourceManager/Compute/Stack/Commands.Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs (hunk 1 above)
Comment (bganapa) [reply src/ResourceManager/Compute/Stack/Commands.Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs]: done

File: src/ResourceManager/Compute/Stack/Compute.sln (hunk 2 above)
Comment (bganapa) [reply src/ResourceManager/Compute/Stack/Compute.sln]: fixed

**Comment by markcowl** [comment]
Closing per this week's discussion

**Pull request closed without merging**